nomad/json/encoding.go

```diff
@@ -0,0 +1,48 @@
+package json
```

Comment:
Does this package name really not cause conflicts with `encoding/json` all over the place? :sweat_smile: 

Looks like only `nomad/json/handlers` imports it... should we just combine them into a `nomad/jsonhandler` package and unexport all of the current `nomad/json` exports? There may be some import issues your method gets around that I'm missing.

Reply:
excellent point, it caused one or two problems, though not as many as you would expect; most of our code either uses this with go-msgpack or `encoding/json`.

i'll rename it. 👍

Reply:
i remember, i had them in different packages because package variables are instantiated before package init is run, but i'll refactor this to get rid of package init.